git://src.develer.com / users/arighi/santerno-imx23 / commitdiff
? search:  re
summary | shortlog | log | commit | commitdiff | tree
raw | patch (parent: 33859ff)
kernel: move spi bitbang driver initialization into the mx23evk special init routine
author	Andrea Righi <arighi@develer.com>
Wed, 12 Jan 2011 15:48:31 +0000 (16:48 +0100)
committer	Andrea Righi <arighi@develer.com>
Wed, 12 Jan 2011 15:48:31 +0000 (16:48 +0100)
Signed-off-by: Andrea Righi <arighi@develer.com>

kernel/linux/arch/arm/mach-mx23/device.c		patch | blob | blame | history
kernel/linux/arch/arm/mach-mx23/mx23evk.c		patch | blob | blame | history
diff --git a/kernel/linux/arch/arm/mach-mx23/device.c b/kernel/linux/arch/arm/mach-mx23/device.c
index 3b6d11f..814c4ef 100644 (file)
--- a/kernel/linux/arch/arm/mach-mx23/device.c
+++ b/kernel/linux/arch/arm/mach-mx23/device.c
@@ -670,64 +670,6 @@ static void mx23_init_mmc(void)
 }
 #endif

-#ifdef CONFIG_SPI_BITBANG
-#include <linux/spi/spi.h>
-#include <linux/spi/spi_bitbang.h>
-#include <linux/spi/spi_gpio.h>
-
-#define SPI_GPIO_SCK   MXS_PIN_TO_GPIO(PINID_LCD_RS)
-#define SPI_GPIO_MOSI  MXS_PIN_TO_GPIO(PINID_LCD_WR)
-#define SPI_GPIO_MISO  MXS_PIN_TO_GPIO(PINID_GPMI_RDY3) /* XXX: unused */
-#define SPI_GPIO_CS    MXS_PIN_TO_GPIO(PINID_LCD_CS)
-
-static struct spi_gpio_platform_data spi_bitbang_gpio_data = {
-       .sck = SPI_GPIO_SCK,
-       .mosi = SPI_GPIO_MOSI,
-       .miso = SPI_GPIO_MISO,
-       .num_chipselect = 1,
-};
-
-static struct platform_device spi_bitbang_pdev = {
-        .name = "spi_gpio",
-        .id = 2,
-        .dev = {
-                .platform_data = &spi_bitbang_gpio_data,
-        },
-};
-
-static struct spi_board_info spi_bitbang_board_info[] __initdata = {
-       {
-               .modalias       = "spidev",
-               .max_speed_hz   = 1000,
-               .bus_num        = 2,
-               .chip_select    = 0,
-               .mode = SPI_MODE_1,
-               .controller_data = (void *)SPI_GPIO_CS,
-       },
-};
-
-static void __init mx23_init_spi_bitbang(void)
-{
-       int ret;
-
-       ret = platform_device_register(&spi_bitbang_pdev);
-       if (unlikely(ret < 0)) {
-               printk(KERN_WARNING "%s: registration failed: %d\n",
-                               __func__, ret);
-               return;
-       }
-       ret = spi_register_board_info(spi_bitbang_board_info,
-                               ARRAY_SIZE(spi_bitbang_board_info));
-       if (unlikely(ret < 0))
-               printk(KERN_INFO "%s: spi registration failed: %d\n",
-                               __func__, ret);
-}
-#else /* !CONFIG_SPI_BITBANG */
-static inline void mx23_init_spi_bitbang(void)
-{
-}
-#endif /* CONFIG_SPI_BITBANG */
-
 #if defined(CONFIG_SPI_MXS) || defined(CONFIG_SPI_MXS_MODULE)
 static struct resource ssp1_resources[] = {
        {
@@ -1003,7 +945,6 @@ int __init mx23_device_init(void)
        mx23_init_rtc();
        mx23_init_dcp();
        mx23_init_ssp1();
-       mx23_init_spi_bitbang();
        mx23_init_gpmi_nfc();
        mx23_init_spdif();
        mx23_init_lcdif();
diff --git a/kernel/linux/arch/arm/mach-mx23/mx23evk.c b/kernel/linux/arch/arm/mach-mx23/mx23evk.c
index 6ce1583..9665853 100644 (file)
--- a/kernel/linux/arch/arm/mach-mx23/mx23evk.c
+++ b/kernel/linux/arch/arm/mach-mx23/mx23evk.c
@@ -116,11 +116,71 @@ int get_evk_board_version()
 }
 EXPORT_SYMBOL_GPL(get_evk_board_version);

+#ifdef CONFIG_SPI_BITBANG
+#include <linux/spi/spi.h>
+#include <linux/spi/spi_bitbang.h>
+#include <linux/spi/spi_gpio.h>
+
+#define SPI_GPIO_SCK   MXS_PIN_TO_GPIO(PINID_LCD_RS)
+#define SPI_GPIO_MOSI  MXS_PIN_TO_GPIO(PINID_LCD_WR)
+#define SPI_GPIO_MISO  MXS_PIN_TO_GPIO(PINID_GPMI_RDY3) /* XXX: unused */
+#define SPI_GPIO_CS    MXS_PIN_TO_GPIO(PINID_LCD_CS)
+
+static struct spi_gpio_platform_data spi_bitbang_gpio_data = {
+       .sck = SPI_GPIO_SCK,
+       .mosi = SPI_GPIO_MOSI,
+       .miso = SPI_GPIO_MISO,
+       .num_chipselect = 1,
+};
+
+static struct platform_device spi_bitbang_pdev = {
+        .name = "spi_gpio",
+        .id = 2,
+        .dev = {
+                .platform_data = &spi_bitbang_gpio_data,
+        },
+};
+
+static struct spi_board_info spi_bitbang_board_info[] __initdata = {
+       {
+               .modalias       = "spidev",
+               .max_speed_hz   = 1000,
+               .bus_num        = 2,
+               .chip_select    = 0,
+               .mode = SPI_MODE_1,
+               .controller_data = (void *)SPI_GPIO_CS,
+       },
+};
+
+static void __init mx23_init_spi_bitbang(void)
+{
+       int ret;
+
+       ret = platform_device_register(&spi_bitbang_pdev);
+       if (unlikely(ret < 0)) {
+               printk(KERN_WARNING "%s: registration failed: %d\n",
+                               __func__, ret);
+               return;
+       }
+       ret = spi_register_board_info(spi_bitbang_board_info,
+                               ARRAY_SIZE(spi_bitbang_board_info));
+       if (unlikely(ret < 0))
+               printk(KERN_INFO "%s: spi registration failed: %d\n",
+                               __func__, ret);
+}
+#else /* !CONFIG_SPI_BITBANG */
+static inline void mx23_init_spi_bitbang(void)
+{
+}
+#endif /* CONFIG_SPI_BITBANG */
+
+
 static void __init mx23evk_device_init(void)
 {
        /* Add mx23evk special code */
        i2c_device_init();
        spi_device_init();
+       mx23_init_spi_bitbang();
        mx23evk_init_adc();
 }
